### PR TITLE
CD-6992 feat(cli): create multiple protections at once

### PR DESCRIPTION
**Pull request by vitorveiga**

### CODE REVIEW

File: packages/jscrambler-cli/src/mutations.js [hunk 1]
```diff
@@ -387,6 +388,28 @@ export function createApplicationProtection(
...
+   ${createProtectionDefaultFragments}
+  }
+`;
+
+export function createApplicationProtections(
+  applicationId,
+  _fragments = createProtectionsDefaultFragments,
+  _options,
+  args,
+  {mutationName = 'createApplicationProtections', mutationInputType = 'ApplicationProtectionsCreate'} = {}
```
Comment (luisreis-jscr) [inline packages/jscrambler-cli/src/mutations.js:402]: Can't we hard-code the mutationName here?

**Review by luisreis-jscr — CHANGES_REQUESTED** [review]

File: packages/jscrambler-cli/README.md [hunk 2]
```diff
@@ -129,6 +129,7 @@ Options:
   --debugMode                             Protect in debug mode
   --skip-sources                          Prevent source files from being updated
   --force-app-environment <environment>   (version 7.1 and above) Override application's environment detected automatically. Possible values: node,browser,isomorphic,automatic
+  -n <number>                             (version 7.2 and above) Create multiple protections at once.
```
Comment (luisreis-jscr) [inline packages/jscrambler-cli/README.md:132]: Seems unfortunate to use such a good shortcut (-n) on a very niche feature.

Comment (vitorveiga) [reply packages/jscrambler-cli/README.md:132]: it was a unanimous decision :)

Comment (luisreis-jscr) [reply packages/jscrambler-cli/README.md:132]: Can we at least *also* have a long version (like all the other options with a short version)?

File: packages/jscrambler-cli/src/index.js [hunk 3]
```diff
@@ -427,104 +431,132 @@ export default {
...
+        )
         .finally(() => process.exit(1));
-    }
+    };
 
     process.once('SIGINT', onExitCancelProtection)
       .once('SIGTERM', onExitCancelProtection);
 
-    const protection = await this.pollProtection(
+    const _protections = await this.pollProtections(
```
[the remark this answers is not in the document]
Comment (vitorveiga) [reply packages/jscrambler-cli/src/index.js]: fixed

File: packages/jscrambler-cli/src/index.js [hunk 4]
```diff
@@ -427,104 +431,132 @@ export default {
...
 
-    await unzip(download, filesDest || destCallback, stream);
+    console.log(`Protections stored in ${filesDest}/[protection-id]`)
 
-    if (debug) {
-      console.log('Finished unzipping files');
-    }
+    await _protections.reduce((acc, protection) =>
+      acc.then(() => handleProtection(protection, {outPrefix: `/${protection._id}/`, printProtectionId: false}))
+         .catch(e => console.error(e)), Promise.resolve()
```
[the remark this answers is not in the document]
Comment (vitorveiga) [reply packages/jscrambler-cli/src/index.js]: fixed

File: packages/jscrambler-cli/src/index.js [hunk 5]
```diff
@@ -929,6 +961,51 @@ export default {
...
+      const applicationProtections = await this.withRetries(
+        () => this.getApplicationProtections(
+          client,
+          applicationId,
+          {protectionIds},
+          fragments.applicationProtection,
+          ["$protectionIds: [String]"]
+        )
+      );
+      //console.log(applicationProtections.data.applicationProtections[0]);
```
[the remark this answers is not in the document]
Comment (vitorveiga) [reply packages/jscrambler-cli/src/index.js]: fixed

File: packages/jscrambler-cli/src/mutations.js (hunk 1 above)
Comment (vitorveiga) [reply packages/jscrambler-cli/src/mutations.js]: fixed

File: packages/jscrambler-cli/README.md (hunk 2 above)
Comment (vitorveiga) [reply packages/jscrambler-cli/README.md:132]: we agree on just having "-n" for now

**Pull request merged**